app/models/custom_attribute.rb

```diff
@@ -1,4 +1,5 @@
 class CustomAttribute < ApplicationRecord
+  ALLOWED_API_FIELD_TYPES = %w(DateTime Time Date).freeze
```

Comment:
This validation is only for the API because the fields which are auto supported in JSON are number, strings, boolean, hence there is no way to insert a date through the rest api (ruby datetime parsing is too permissive), that's why those are the only supported types through the API (the rest of the types don't need to be mentioned).